src/dotnet/commands/dotnet-nuget/NuGetCommandParser.cs

```diff
@@ -5,74 +5,52 @@
-                                                        Accept.ExactlyOneArgument()
-                                                              .With(name: "timeout")),
-                                          Create.Option("-k|--api-key", "The API key for the server.",
-                                                        Accept.ExactlyOneArgument()
-                                                              .With(name: "apiKey")),
-                                          Create.Option("-sk|--symbol-api-key", "The API key for the symbol server.",
-                                                        Accept.ExactlyOneArgument()
-                                                              .With(name: "apiKey")),
-                                          Create.Option("-d|--disable-buffering",
-                                                        "Disable buffering when pushing to an HTTP(S) server to decrease memory usage."),
-                                          Create.Option("-n|--no-symbols",
-                                                        "If a symbols package exists, it will not be pushed to a symbols server.")));
+            Create.Command(
+                "nuget",
+                Parser.NoDescription,
+                Create.Option("-h|--help", Parser.NoDescription),
+                Create.Option("--version", Parser.NoDescription),
+                Create.Option("-v|--verbosity", Parser.NoDescription, Accept.ExactlyOneArgument()),
+                Create.Command(
+                    "delete",
+                    Parser.NoDescription,
+                    Accept.OneOrMoreArguments(),
+                    Create.Option("-h|--help", Parser.NoDescription),
+                    Create.Option("--force-english-output", Parser.NoDescription),
+                    Create.Option("-s|--source", Parser.NoDescription, Accept.ExactlyOneArgument()),
+                    Create.Option("--non-interactive", Parser.NoDescription),
+                    Create.Option("-k|--api-key", Parser.NoDescription, Accept.ExactlyOneArgument()),
+                    Create.Option("--no-service-endpoint", Parser.NoDescription)),
+                Create.Command(
+                    "locals",
+                    Parser.NoDescription,
+                    Accept.AnyOneOf(
+                        "all",
+                        "http-cache",
+                        "global-packages",
+                        "temp"),
+                    Create.Option("-h|--help", Parser.NoDescription),
+                    Create.Option("--force-english-output", Parser.NoDescription),
+                    Create.Option("-c|--clear", Parser.NoDescription),
+                    Create.Option("-l|--list", Parser.NoDescription)),
+                Create.Command(
+                    "push",
+                    Parser.NoDescription,
+                    Accept.OneOrMoreArguments(),
```

Comment:
This is to sync with nuget's command structure? If it is different we will block valid commands.

Reply:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PushCommand.cs#L75 it is expected

Reply:
That's correct.  It's to sync with what the `push` command actually accepts.  This prevents us from sending telemetry for commands that shouldn't successfully parse (i.e. `dotnet nuget push` is currently an error for a missing argument, yet we send telemetry for it because the fake parser previously accepted no arguments).  This is also why the test was updated.